co-noir/co-noir/src/bin/co-noir.rs

```diff
@@ -253,6 +262,62 @@ fn run_split_input(config: SplitInputConfig) -> color_eyre::Result<ExitCode> {
     Ok(ExitCode::SUCCESS)
 }
 
+#[instrument(skip(config))]
+fn run_merge_input_shares(config: MergeInputSharesConfig) -> color_eyre::Result<ExitCode> {
+    let inputs = config.inputs;
+    let protocol = config.protocol;
+    let out = config.out;
+
+    if protocol != MPCProtocol::REP3 {
+        return Err(eyre!(
+            "Only REP3 protocol is supported for splitting/merging inputs"
+        ));
+    }
+
+    if inputs.len() < 2 {
+        return Err(eyre!("Need at least two input shares to merge"));
+    }
+    for input in &inputs {
+        file_utils::check_file_exists(input)?;
+    }
+
+    let start = Instant::now();
+    let input_shares = inputs
+        .iter()
+        .map(|input| {
+            // parse input shares
+            let input_share_file =
+                BufReader::new(File::open(input).context("while opening input share file")?);
+            let input_share: WitnessMap<Rep3PrimeFieldShare<ark_bn254::Fr>> =
+                bincode::deserialize_from(input_share_file)
+                    .context("while deserializing input share")?;
+            color_eyre::Result::<_>::Ok(input_share)
+        })
+        .collect::<Result<Vec<_>, _>>()?;
+    let mut result = WitnessMap::new();
+    let mut offset = 0;
+
+    for input_share in input_shares.into_iter() {
+        let mut els = 0;
+        for (wit, share) in input_share.into_iter() {
+            let wit = wit.0 + offset;
+            result.insert(wit.into(), share);
```

Comment:
we should probably emit at least a warning if both parties provide the same input

Reply:
also what happens now if the input is not complete, does the witness extension fail nicely then?

Reply:
there is at this point not really a way to check anymore